plugins/catalog/src/components/ComponentRemovalDialog/ComponentRemovalDialog.tsx

```diff
@@ -101,10 +104,25 @@ export const ComponentRemovalDialog: FC<ComponentRemovalDialogProps> = ({
         ) : null}
       </DialogContent>
       <DialogActions>
-        <Button onClick={onClose}>Cancel</Button>
+        <Button onClick={onClose} color="primary">
+          Cancel
+        </Button>
         <Button
           disabled={!!(loading || error)}
-          onClick={onConfirm}
+          onClick={async () => {
+            const uid = entity.metadata?.uid;
+            if (uid) {
+              try {
+                await catalogApi.removeEntityByUid(uid);
+              } catch (err) {
+                alertApi.post({ message: err.message });
+              }
+            } else {
+              alertApi.post({ message: `No entity with UID ${uid}` });
```

Comment:
yeah no remove this, it's not particularly true either (if no entity with that uid was found, you would have `entity === undefined` instead or something - and the props for this component require a proper Entity to be given)